File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen_pair = dict()

        for i, num in enumerate(nums):
            if str(num) not in seen_pair.keys():
                seen_pair[str(target - num)] = i
            else:
                return [seen_pair[str(num)], i]

File: Data Structures & Algorithms/remove-element/submission-0.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        val target = `val`
        val n = nums.size
        var last = n - 1
        var count = nums.count { it == target }
        val k = n - count

        for (i in 0 until k) {
            if (nums[i] == target) {
                while (nums[last] == target) last--
                nums[i] = nums[last--]
            }
        }

        return k
    }
}
